veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,259 @@
+
+## Overview
+
+Introduce a Hypervisor Abstraction Layer that lets KubeVirt plug in multiple hypervisor backends through a consistent contract, while keeping today's KVM-first behavior unchanged. To start, we scope the contract to device selection, domain tweaks, validation and mutation so as to provide a flexible base that can evolve without introducing a major refactor. 
+
+## Motivation
+
+- KubeVirt currently hard-codes KVM/QEMU assumptions throughout virt-launcher, virt-handler, virt-controller, and node preparation scripts.
+- Platform teams that rely on alternative accelerators or hypervisor stacks face invasive forks to replace `/dev/kvm`, libvirt domain types, or resource scheduling hints.
+- A scoped abstraction keeps the project approachable for contributors while unlocking new hardware backends.
+
+## Goals
+
+- Provide an explicit `Hypervisor` interface that supplies device requirements to schedule against plus domain defaults and mutators, avoiding invasive changes to existing components.
+- Resolve the active hypervisor early and feed it through the virt-launcher converter so hypervisor-specific behavior stays localized.
+- Let components request allocatable device resources declared by the hypervisor configuration, avoiding new scheduling primitives.
+- Make it simple for downstreams to implement new hypervisors by following a documented contract.
+
+## Non Goals
+
+- Deliver a full implementation of any specific new hypervisor backend.
+- Redesign the VirtualMachineInstance API schema beyond additive fields.
+- Mandate new observability requirements; telemetry hooks remain optional.
+
+## Definition of Users
+
+- Cluster administrators who need to bootstrap KubeVirt on hardware that exposes alternative virtualization devices.
+- Platform vendors integrating proprietary or emerging hypervisor stacks with KubeVirt.
+- Upstream contributors maintaining virt-launcher, virt-controller, and virt-handler.
+
+## User Stories
+
+1. As a cluster administrator, I can declare a non-KVM hypervisor as the cluster default, and VMI pods schedule only on nodes that expose its required devices.
+2. As a platform engineer, I can supply hypervisor-specific libvirt defaults and mutators without forking the virt-launcher domain converter.
+3. As an upstream maintainer, I know exactly where to add validation, testing, and documentation when a new hypervisor is introduced.
+
+## Repos
+
+- [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+- [kubevirt/enhancements](https://github.com/kubevirt/enhancements) (this VEP)
+
+## Design
+
+### Hypervisor Interface
+
+```go
+// pkg/hypervisor/hypervisor.go
+
+type DeviceRequirement struct {
+  Resource string
+  Count    int64
+  Optional bool
+}
+
+type DomainProfile struct {
+  Type string
+  XMLNS string
+  Mutators []DomainMutator
+}
+
+type DomainMutator interface {
+  Apply(*api.Domain, *v1.VirtualMachineInstance) error
+}
+
+type ValidationResult = field.ErrorList
+
+type Hypervisor interface {
```

Comment:
I think the main functions we are missing here are 
```
AdjustResources(vmi *v1.VirtualMachineInstance, additionalOverheadRatio *string) error
HandleHousekeeping(vmi *v1.VirtualMachineInstance, domain *api.Domain) error
GetMemoryOverhead(vmi *v1.VirtualMachineInstance, arch string, additionalOverheadRatio *string) resource.Quantity
``` perhaps more hypervisor specific functions